lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/Transactions.hs

```diff
@@ -494,10 +492,6 @@ fixtureExternalTx ctx toSend = do
              , mkSeqState @n (rootXPrv, pwd) defaultAddressPoolGap
              )
 
-encodeTx :: (Tx, [TxWitness]) -> FragmentSpec -> Base -> Text
-encodeTx (tx, wits) msgType base =
-    T.decodeUtf8 $ convertToBase base $ BL.toStrict $ encode (tx, wits) msgType
-  where
-    encode ((Tx _ inps' outs'), wits') msgType' = runPut
-        $ withHeader msgType'
-        $ putSignedTx inps' outs' wits'
+txToBase :: SignedTxBinary -> Base -> Text
+txToBase (SignedTxBinary bin) base =
+    T.decodeUtf8 $ convertToBase base bin
```

Comment:
I renamed `encodeTx` to `txToBase` because it doesn't encode anything to binary anymore. Might be better names.

Reply:
May I suggest instead to make `SignedTxBinary` an instance of [ByteArrayAccess](https://hackage.haskell.org/package/memory-0.15.0/docs/Data-ByteArray.html#t:ByteArrayAccess) (`GeneralizedNewTypeDeriving` + `DerivingStrategy`) and to simply rely on [hex](https://input-output-hk.github.io/cardano-wallet/haddock/edge/cardano-wallet-core-2019.11.6/Cardano-Wallet-Primitive-AddressDerivation.html#v:hex) (for lack of a better place for this helper for now) to do conversion to hexadecimal? 

This will allow for more code re-use and less helper functions to be used.

Reply:
We still need the `T.decodeUtf8` and the ability to encode to an arbitrary base. We expect `postExternalTxEp` to reject hex-encoded transactions and accept base64-encoded ones. It seems like this helper is still good to have then. It is called 6 times.

An `ByteArrayAccess` sounds cool though, and removes the need for `(SignedTxBinary bin)`. Added.

Reply:
Actually, we didn't need `T.decodeUtf8` everywhere. I removed the helper 👍 

b088524bef896e7f01085e8d9de1f31e078d85be